docs/api.rst

```diff
@@ -2850,15 +2871,170 @@ The details of each asset can contain the following keys:
                   "cryptocompare":"VET",
                   "coingecko":"vet",
                   "protocol":"None"
+              }
+          ],
+          "message": ""
+      }
+
+   :resjson object result: A list of objects that contain the asset details.
+   :statuscode 200: Assets successfully queried.
+   :statuscode 500: Internal rotki error
+
+
+Get asset identifiers mappings
+================================
+
+.. http:post:: /api/(version)/assets/mappings
+
+   Doing a GET on the assets mappings endpoint will return a list of objects containing an asset's name, symbol and identifier.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      POST /api/1/assets/mappings HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {
+        "identifiers": [
+            "eip155:1/erc20:0xB6eD7644C69416d67B522e20bC294A9a9B405B31",
+            "DCR",
+            "eip155:1/erc20:0xcC4eF9EEAF656aC1a2Ab886743E98e97E090ed38"
+        ]
+      }
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+          "result": [
+              {
+                  "identifier": "eip155:1/erc20:0xB6eD7644C69416d67B522e20bC294A9a9B405B31",
+                  "name": "0xBitcoin",
+                  "symbol": "0xBTC"
               },
-          },
+              {
+                  "identifier": "DCR",
+                  "name": "Decred",
+                  "symbol": "DCR"
+              },
+              {
+                  "identifier": "eip155:1/erc20:0xcC4eF9EEAF656aC1a2Ab886743E98e97E090ed38",
+                  "name": "DigitalDevelopersFund",
+                  "symbol": "DDF"
+              }
+          ],
+          "message": ""
+      }
+
+   :resjson object result: A list of objects that contain the asset mappings.
+   :statuscode 200: Assets successfully queried.
+   :statuscode 400: One of the identifiers are not valid. Provided JSON is in some way malformed.
+   :statuscode 500: Internal rotki error.
+
+Search for assets
+===================
+
+.. http:post:: /api/(version)/assets/search
+
+   Doing a POST on the assets search endpoint will return a list of objects containing an asset's name, symbol and identifier based on the search keyword provided.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      POST /api/1/assets/search HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {
+        "substring_search": "bitcoin",
+        "limit": 50
```

Comment:
you are also missing another thing we discussed. This endpoint should specify if the search should search by name or by symbol. It's also missing from the spec from what I saw.

The current implementation in 1.25.3 does a fuzzy search in both name and symbol (so the levensthein option) but with the simple DB query we would need to specify either name or symbol